### ISSUE TITLE
ORA-12505 when accessing via JDBC

### ISSUE THREAD

**Issue by hallatech** [oracle/docker-images#692]
I've setup 2 containers, 1 being the DB using SE2, the second is using JDBC to connect to the database.

The JDBC connection fails with
ORA-12505, TNS:listener does not currently know of SID given in connect descriptor

All the other connections work as documented:
- Accessing the ORCLCDB via sqlplus in the DB container
- Accessing the ORCLPDB1 via sqlplus in the DB container
- Accessing https://localhost:5500 on the host

JDBC connection details jdbc:oracle:thin:@database:1521:orclpdb1,username=sys as sysdba,password=Admin123

This is the same for orclpdb1 or ORCLPDB1

Here is the alert.log output:
`/opt/oracle/diag/tnslsnr/d20f267a00be/listener/alert/log.xml`
<msg time='2017-12-28T13:07:24.307+00:00' org_id='oracle' comp_id='tnslsnr'
 type='UNKNOWN' level='16' host_id='d20f267a00be'
 host_addr='172.18.0.3'>
 <txt>28-DEC-2017 13:07:24 * (CONNECT_DATA=(SID=orclpdb1)(CID=(PROGRAM=JDBC Thin Client)(HOST=__jdbc__)(USER=atg))) * (ADDRESS=(PROTOCOL=tcp)(HOST=172.18.0.2)(PORT=57394)) * establish * orclpdb1 * 12505
 </txt>
</msg>
<msg time='2017-12-28T13:07:24.307+00:00' org_id='oracle' comp_id='tnslsnr'
 type='UNKNOWN' level='16' host_id='d20f267a00be'
 host_addr='172.18.0.3'>
 <txt>TNS-12505: TNS:listener does not currently know of SID given in connect descriptor
 </txt>
</msg>

AS you can see I can access in the container directly:
[oracle@d20f267a00be ~]$ sqlplus sys/Admin123@//localhost:1521/ORCLPDB1 as sysdba

SQL*Plus: Release 12.1.0.2.0 Production on Thu Dec 28 13:24:10 2017

Copyright (c) 1982, 2014, Oracle.  All rights reserved.

Connected to:
Oracle Database 12c Standard Edition Release 12.1.0.2.0 - 64bit Production

SQL>

[oracle@d20f267a00be ~]$ cat /opt/oracle/product/12.1.0.2/dbhome_1/network/admin/listener.ora 
LISTENER = 
(DESCRIPTION_LIST = 
  (DESCRIPTION = 
    (ADDRESS = (PROTOCOL = IPC)(KEY = EXTPROC1)) 
    (ADDRESS = (PROTOCOL = TCP)(HOST = 0.0.0.0)(PORT = 1521)) 
  ) 
) 

DEDICATED_THROUGH_BROKER_LISTENER=ON
DIAG_ADR_ENABLED = off

[oracle@d20f267a00be ~]$ cat /opt/oracle/product/12.1.0.2/dbhome_1/network/admin/tnsnames.ora 
ORCLCDB=localhost:1521/ORCLCDB
ORCLPDB1= 
(DESCRIPTION = 
  (ADDRESS = (PROTOCOL = TCP)(HOST = 0.0.0.0)(PORT = 1521))
  (CONNECT_DATA =
    (SERVER = DEDICATED)
    (SERVICE_NAME = ORCLPDB1)
  )
)

The only overrides I'm using:
database:
  environment:
    - ORACLE_PWD=Admin123
    - ORACLE_CHARACTERSET=UTF8

This comes from the database built using: database:12.1.0.2-se2

Is there anything else that I would need to configure for the JDBC connection to work - I can't see anything obvious. Any command I want to run via JDBC works directly in sqlplus using ORCLPDB1.

**Comment by gvenzl** [comment]
Hi @hallatech, in your JDBC connect string you are confusing the `SERVICE_NAME` with the `SID` of the database.

You are trying to connect to the SID `orclpdb1` instead of the service name: `jdbc:oracle:thin:@database:1521:orclpdb1,username=sys as sysdba,password=Admin123`

 This is because you are using a `:` after the port and not a `/` like you also do below with SQLPlus: `sqlplus sys/Admin123@//localhost:1521/ORCLPDB1 as sysdba`

The SID of your container is `ORCLCDB` which is the container database instance itself. Your PDB `ORCLPDB1` is exposed via a service (as is your container database as well).

In short, try to change your JDBC connect string to `jdbc:oracle:thin:@database:1521/orclpdb1,username=sys as sysdba,password=Admin123` and try again.

**Comment by hallatech** [comment]
You're correct thank you! We were still using the XE JDBC Url template and simply need to switch over to the SE2 one, simply requiring the '/' instead of ':'
Problem solved.

**Comment by gvenzl** [comment]
I highly recommend always using `/` going forward from now on. Services provide a lot more benefits than connecting directly to the SID and, with Multitenant, it's the only way to connect to the PDB.
